core/templates/dev/head/domain/story_viewer/StoryPlaythroughObjectFactory.ts

```diff
@@ -0,0 +1,76 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating and mutating instances of frontend
+ * story playthrough domain objects.
+ */
+
+oppia.factory('StoryPlaythroughObjectFactory', [
+  'ReadOnlyStoryNodeObjectFactory', function(ReadOnlyStoryNodeObjectFactory) {
```

Comment:
Forgot to add require for this factory?

Reply:
Oh, will add it. This was called from a directive which had imported the factory, so no error came.

Reply:
Done